### PR TITLE
[ticket/13804] Make template's user dependency optional

### PR DESCRIPTION
**Pull request by MateBartus**
[PHPBB3-13804](https://tracker.phpbb.com/browse/PHPBB3-13804)

### CODE REVIEW

File: phpBB/phpbb/template/exception/user_service_not_found.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\template\exception;
+
+/**
+ * This exception is thrown when the user object was not set but it is required by the called method
+ */
+class user_service_not_found extends \phpbb\exception\runtime_exception
```
Comment (Nicofuma) [inline phpBB/phpbb/template/exception/user_service_not_found.php]: user_object_not_available or user_not_available

File: phpBB/phpbb/template/twig/twig.php [hunk 2]
```diff
@@ -344,11 +353,20 @@ protected function get_template_vars()
...
 			array(
 				'definition'	=> new \phpbb\template\twig\definition(),
-				'user'			=> $this->user,
 				'loops'			=> $context_vars, // To get loops
 			)
 		);
 
+		if ($this->user instanceof \phpbb\user)
+		{
+			$vars = array_merge(
```
Comment (Nicofuma) [inline phpBB/phpbb/template/twig/twig.php]: `$vars['user'] = $this->user;` ?

Comment (MateBartus) [inline phpBB/phpbb/template/twig/twig.php]: My intention was to be able to keep `$context_vars['.'][0]['user']['something']` if that is something that is used. However it is not required I fix it

Comment (Nicofuma) [inline phpBB/phpbb/template/twig/twig.php]: Actually I don't see the difference....

Comment (Nicofuma) [inline phpBB/phpbb/template/twig/twig.php]: ok ^^

**Pull request merged**